test/test_reader.cpp

```diff
@@ -345,10 +345,10 @@ TEST(GoosReader, FromFile) {
 TEST(GoosReader, TextDb) {
   // very specific to this particular test file, but whatever.
   Reader reader;
-  auto result = reader.read_from_file({"test", "test_data", "test_reader_file0.gc"})
-                    .as_pair()
-                    ->cdr.as_pair()
-                    ->car;
+  auto& result = reader.read_from_file({"test", "test_data", "test_reader_file0.gc"})
```

Comment:
This one isn't safe to be a reference - the reader returns an `Object` by value.